cli/runner/cmd/requests.go

```diff
@@ -82,6 +86,28 @@ func postSourceFile(server string, endpoint string, filepath string, langCheck s
 	req.Header.Add("Content-Type", "application/json")
 
 	//TODO: implement CLI check language logic before adding cookie
+	if langCheck == "implicit" {
+		langs, err := getLangListJSON(server, LANG_ENDPOINT)
+		if err != nil {
+			return nil, err
+		}
+		ext := extractExtension(filepath)
+
+		//loop through supported languages
+		//if file extension map at supported language == ext set langcheck
+		//else throw err
+
+		for _, j := range langs.Languages {
```

Comment:
Oh, now I see there's a missing "reverse" map that would've made this easier. How about instead adding a new map in `enginer/coderunner/runner.go` so you don't have to do the loop?

```go
var ExtensionToLanguageMap = map[string]Language {
	"cpp": CPP11,
    "sh" : SHELL,
    "py" : PYTHON3,
}
```

It's somewhat duplicated work but I think we'll also want to use it to provide more metadata from the `langs` endpoint later. Then, all should have to do too get the language is `langCheck = coderunner.ExtensionToLanguageMap[ext]`.